core/domain/user_services_test.py

```diff
@@ -129,3 +133,236 @@ def test_get_user_id_from_username(self):
+        rights_manager.release_ownership_of_exploration(self.admin_id,
+            self.EXP_ID)
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.login(self.EDITOR_EMAIL)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+
+        exp_services.update_exploration(
+            self.editor_id, self.EXP_ID, [{
+            'cmd': 'edit_state_property',
+            'state_name': self.init_state_name,
+            'property_name': 'widget_id',
+            'new_value': 'MultipleChoiceInput'
+            }], 'commit')
+
+        self.assertIsNotNone(user_services.get_user_settings(self.editor_id)
+            .first_contribution_msec)
+
+    def test_contribution_msec_updates_on_unpublished_explorations(self):
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.login(self.ADMIN_EMAIL)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.login(self.EDITOR_EMAIL)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.admin_id, end_state_name='End')
+        self.init_state_name = exploration.init_state_name
+
+        # Test that saving an exploration does not update first contribution
+        # time.
+        self.assertIsNone(user_services.get_user_settings(self.admin_id)
+            .first_contribution_msec)
+
+        # Test that commit to unpublished exploration does not update
+        # contribution time.
+        exp_services.update_exploration(
+            self.admin_id, self.EXP_ID, [{
+            'cmd': 'edit_state_property',
+            'state_name': self.init_state_name,
+            'property_name': 'widget_id',
+            'new_value': 'MultipleChoiceInput'
+            }], '')
+        self.assertIsNone(user_services.get_user_settings(self.admin_id)
+            .first_contribution_msec)
+
+        # Test that another user who commits to unpublished exploration does not
+        # have updated first contribution time.
+        rights_manager.assign_role_for_exploration(
+            self.admin_id, self.EXP_ID, self.editor_id, 'editor')
+        exp_services.update_exploration(
+            self.editor_id, self.EXP_ID, [{
+            'cmd': 'rename_state',
+            'old_state_name': feconf.DEFAULT_INIT_STATE_NAME,
+            'new_state_name': u'¡Hola! αβγ',
+            }], '')
+        self.assertIsNone(user_services.get_user_settings(self.editor_id)
+            .first_contribution_msec)
+
+        # Test that after an exploration is published, all contributors have
+        # updated first contribution time.
+        rights_manager.publish_exploration(self.admin_id, self.EXP_ID)
+        exploration_rights = rights_manager.get_exploration_rights(self.EXP_ID)
+        for owner_id in exploration_rights.owner_ids:
```

Comment:
I think it's better, here, to check admin_id, editor_id separately. Also, editor_id shouldn't get credit, right? Worth checking that, perhaps.